foolbox/tests/test_attacks_adef_attack.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+
+from foolbox.attacks import ADefAttack as Attack
+
+
+def test_attack_pytorch(bn_adversarial_pytorch):
+    adv = bn_adversarial_pytorch
+    attack = Attack()
+    attack(adv)
+    assert adv.image is not None
+    assert adv.distance.value < np.inf
+
+
+def test_targeted_attack_pytorch(bn_targeted_adversarial_pytorch):
+    adv = bn_targeted_adversarial_pytorch
+    attack = Attack()
+    attack(adv)
+    assert adv.image is None
+    assert adv.distance.value == np.inf
+
+
+def test_attack(bn_adversarial):
+    adv = bn_adversarial
+    attack = Attack()
+    attack(adv)
+    assert adv.image is not None
+    assert adv.distance.value < np.inf
+
+
+def test_attack_gl(gl_bn_adversarial):
+    adv = gl_bn_adversarial
+    attack = Attack()
+    attack(adv)
+    assert adv.image is None
+    assert adv.distance.value == np.inf
+
+
+def test_targeted_attack(bn_targeted_adversarial):
+    adv = bn_targeted_adversarial
+    attack = Attack()
+    attack(adv)
+    assert adv.image is None
```

Comment:
this test should make sure that targeted attacks work, not that they fail

Reply:
alles klar, ich ändere das. ich habs von test_attacks_deepfool kopiert, line 38 :)